tools/autograd/gen_variable_type.py

```diff
@@ -311,6 +312,39 @@
 ${return_type} ${api_name}(${type_method_formals}); // ${schema_string}
 """)
 
+# ProfiledType templates
+PROFILE_DISPATCH_TO_NON_VAR_TYPE_WITH_RETURN_VALUES = CodeTemplate("""\
+{
+    AutoNonProfileTypeMode non_prof_type_mode;
+    RECORD_FUNCTION("${name}", std::vector<c10::IValue>({${input_names}}), Node::peek_at_next_sequence_nr());
+    return ${base_type_call};
```

Comment:
I think it's OK to unconditionally use this template, as using a return in a void function with an expression is fine if the expression itself returns void. Citation: https://en.cppreference.com/w/cpp/language/return